tests/test_partition.py

```diff
@@ -1,11 +1,13 @@
 # Copyright 2023 MosaicML Streaming authors
 # SPDX-License-Identifier: Apache-2.0
 
+import pytest
```

Comment:
Can we also have a test that validates that no sample duplication and no drop samples when doing a resumption? You might have to track the sample ID and check if there is any sample duplication or drop samples!

Reply:
I'll modify the test to do the following:
* If the NCN and PN value was supported by the `orig` partitioning algo, then make sure the output of `relaxed` partition is the same as the `orig` partition.
* If the NCN and PN value was not supported by the `orig` partitioning algo, then make sure that the output of the `relaxed` partition, when split up by global batches, is **the exact same** as the initial sample partition. `relaxed` partitioning is written so that this condition must be true. This means that there is no additional sample duplication and no additional dropped samples compared to the initial sample partition when doing a resumption.